lib/PWC/Utils/DataMigration.pm

```diff
@@ -1,63 +1,56 @@
-#!/usr/bin/perl
-
-package PerlWeeklyChallenge;
+package PWC::Utils::DataMigration;
 
+use strict; use warnings;
 use Moo;
 use MooX::Options;
-use JSON;
 use File::Find qw(finddepth);
 use File::Basename qw(basename);
 use File::Find::Rule qw(rule);
 use File::Copy::Recursive qw(dircopy);
+use PWC::Utils;
+use PWC::Utils::DataMigration::Option;
 
-option 'project_dir' => (is => 'ro', format => 's', doc => 'Path to the perlweeklychallenge-club folder', default => sub {"."});
+with 'PWC::Utils::DataMigration::Option';
 
 my $restructure_folder;
 my @weekly_challenge_folders;
 
 sub run {
     my ($self) = @_;
-    my $project_dir = $self->{project_dir};
-    $restructure_folder = "$project_dir/restructure";
+    my $project_dir = $self->project_dir;
+    $restructure_folder = "restructure";
 
     mkdir $restructure_folder or die "Failed to create folder $restructure_folder: $!";
 
-    my $members_json_file = "$project_dir/members.json";
     @weekly_challenge_folders = rule(
         directory =>
         maxdepth => 1,
         name => qr/challenge-\d{3}/,
     )->in($project_dir);
 
-    my $members_hash_ref = read_data($members_json_file);
+    my $members_json_file = "$project_dir/members.json";
+    my $members_hash_ref = PWC::Utils::read_data($members_json_file);
     my @member_nicks = keys %$members_hash_ref;
 
-    foreach my $nick (@member_nicks)
+    my $guests_json_file = "$project_dir/guests.json";
+    my $guests_hash_ref = PWC::Utils::read_data($guests_json_file);
+    my @guest_nicks = keys %$guests_hash_ref;
+
+    my @all_nicks = (@member_nicks, @guest_nicks);
+
+    foreach my $nick (@all_nicks)
     {
         populate_member_folder($nick);
     }
-
-    print "Done";
-}
-
-sub read_data {
-    my ($filename) = @_;
-
-    my $json_text = do {
-        open(my $json_fh, "<:encoding(UTF-8)", $filename)
-            or die("Can't open \$filename\": $!\n");
-        local $/;
-        <$json_fh>
-    };
-
-    return JSON->new->allow_nonref->utf8(1)->decode($json_text);
 }
 
 sub populate_member_folder {
     my ($nick) = @_;
```

Comment:
check if $nick is defined doing something like below:
       return unless defined $nick;